docs/zkapps/tutorials/11-advanced-account-updates.mdx

```diff
@@ -11,48 +20,42 @@ zkApps can now be deployed to Berkeley Testnet.
 
-An earlier tutorial covered how each token is associated with a manager account and how the manager account determines rules for token minting, burning, and transfer.
+Internally, it's more general. The zkApp manager account for a token actually controls all properties of token accounts.
 
-Internally, what is happening is actually a bit more general than that. The Manager Account for a token actually controls all properties of token accounts.
+For an account update to be able to modify state (including sending tokens) on a token account, two things must be true:
 
-## zkApp Manager Accounts and Tokens
+1. The account update must meet the permissions criteria on the zkApp account itself. 
 
-For an Account Update to be able to modify state (including sending tokens) on a token account, two things must be true:
+  If the zkApp Account is set up with the `proof` permission, the account update proof must pass the verification key in the account.
+2. The account update must set `mayUseToken` to a truthy value (a value that evaluates to true in a Boolean context). 
 
-1. The account update must meet the criteria of the permissions on the zkApp account itself. If the zkApp Account is set up with the `proof` permission, the Account Update proof must pass the verification key in the account.
-2. The account update must set `mayUseToken` to a truthy value. However, to be allowed to do so, a transaction must get its account update “approved” by the corresponding manager account. The manager account can approve accounts based on its `access` permission.
+  However, for this to happen, a transaction must get its account update approved by the corresponding zkApp manager account. The zkApp manager account can approve accounts based on its `access` permission.
 
 ## zkApp Token Management
 
-As an example, you can write a token that can approve a zkApp account to send tokens from itself to another zkApp.
+For example, you can write a token that can approve a zkApp account to send tokens from itself to another zkApp.
 
-To implement this, you would have 3 contracts:
+To implement, you need three contracts:
 
-1. `MyToken`: The token contract. On instantiation, serves as the zkAppManagerContract for all instances of the token.
-2. `TokenUser`: A zkApp contract that manages tokens. This one is instantiated on the Mina (default) tokenId.
-3. `TokenHolder`: A zkApp contract that stores tokens on behalf of the `TokenUser` contract. This one is instantiated on the `MyToken` instance TokenId.
+1. `MyToken`: The token contract. On instantiation, it serves as the zkAppManagerContract for all instances of the token.
+2. `TokenUser`: A zkApp contract that manages tokens. This contract is instantiated on the Mina (default) `tokenId`.
+3. `TokenHolder`: A zkApp contract that stores tokens on behalf of the `TokenUser` contract. This contract is instantiated on the `MyToken` instance tokenId.
```

Comment:
@Trivo25  I don't find this file in the https://github.com/o1-labs/docs2/tree/main/examples/zkapps/11-advanced-account-updates/ folder

can you help?

Reply:
Looks like it references this file, @barriebyron: https://github.com/o1-labs/docs2/blob/main/examples/zkapps/11-advanced-account-updates/src/main.ts

I asume this based on the following snippet:
```
const myTokenInstance = new MyToken(myTokenAddr);
const tokenUserInstance = new TokenUser(tokenUserAddr);
const tokenHolderInstance = new TokenHolder(
  tokenUserAddr,
  myTokenInstance.token.id
);
```
...which is referenced in that paragrpah in the docs and found in that `main.ts` file